### ISSUE TITLE
Strange behaviour of `react/jsx-curly-brace-presence`/`react/self-closing-comp`

### ISSUE THREAD

**Issue by dimaMachina** [mdx-js/eslint-mdx#437]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

2.0.5

### Steps to reproduce

````mdx
// test.mdx

```jsx
let a = (
  <App prop="foo">{'bar'}</App> // ❌ error  Curly braces are unnecessary here  react/jsx-curly-brace-presence
)
```

<App prop="foo">{'bar'}</App> // 😬 nothing here
````
and

````mdx
```jsx
let a = <App prop={'foo'}>bar</App> ❌ error  Curly braces are unnecessary here  react/jsx-curly-brace-presence
```

<App prop={'foo'}>bar</App> ❌ error  Curly braces are unnecessary here  react/jsx-curly-brace-presence / error  Empty components are self-closing  react/self-closing-comp
````

### Expected behavior

should report only `react/jsx-curly-brace-presence` errors

### Actual behavior

reports `react/jsx-curly-brace-presence` and `react/self-closing-comp`

**Comment by B2o5T** [comment]
repro https://github.com/B2o5T/mdx-repro

**Comment by u3u** [comment]
Why is this happening? I'm not sure if there are any other rules that could cause strange issues, which makes me fearful of eslint-mdx. 😱

```mdx
❌ Empty components are self-closing eslint(react/self-closing-comp)
<span>test</span>

💡 Fix this react/jsx-sort-props problem
<Image alt="alt" src="src" width="315" height="100" format="svg" />
↓↓↓↓↓
<Image alt="alt" format="src" height="315" src="100" width="svg" />
```

**Comment by JounQin** [comment]
This happens because we don't know whether it's self-closing from mdx AST.

cc @wooorm 

---

@u3u I don't know what's your meaning.

**Comment by u3u** [comment]
@JounQin There are serious issues when used with the `react/self-closing-comp` / `react/jsx-sort-props` rules.
1. `<span>test</span>` Regarded as an empty tag self-closing, it will be fixed to `<span />`, directly losing the content.
2. After automatic sorting of JSX attributes, the content and attributes no longer correspond.

**Comment by JounQin** [comment]
@u3u Please raise a new issue with online runnable reproduction, it seems a different issue from this one, and I've never met.

**Comment by u3u** [comment]
@JounQin Clone this repository <https://github.com/u3u/eslint-config>, and input the above content in any mdx file to reproduce.

**Comment by JounQin** [comment]
@u3u A new issue please, we must track different issues.

**Comment by wooorm** [comment]
> `<App prop="foo">{'bar'}</App> // 😬 nothing here`

MDX is not JSX. There is a difference. `<x>*a*</x>` vs `<x>{'*a*'}</x>` are different.

> This happens because we don't know whether it's self-closing from mdx AST.

Indeed. The AST is supposed to be simple and changeable.

What has self-closing to do with this example? It’s *not* a self-closing JSX element?

**Comment by JounQin** [comment]
> What has self-closing to do with this example? It’s not a self-closing JSX element?

[`react/self-closing-comp`](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/self-closing-comp.md) rule is referenced, so I said it's related to self-closing.

**Comment by wooorm** [comment]
It shouldn’t be in the AST.

ASTs are about simplifying things. Not about presenting everything. Similar to how whether `"` or `'` was used is not in ASTs. That’s what CSTs are.

**Comment by JounQin** [comment]
OK, let's skip this part.

**Comment by JounQin** [comment]
> What has self-closing to do with this example? It’s not a self-closing JSX element?

I think that's because the jsx content is not included or incorrect in the AST, so `react/self-closing-comp` rule does not recognize the jsx content correctly, I'll take a look how can I fix it.

**Comment by JounQin** [comment]
@dimaMachina @u3u

Please help to test the part of `react/self-closing-comp`.

```sh
# yarn 1
yarn add https://pkg.csb.dev/mdx-js/eslint-mdx/commit/d4a9ad99/eslint-mdx https://pkg.csb.dev/mdx-js/eslint-mdx/commit/d4a9ad99/eslint-plugin-mdx
# yarn 2, 3
yarn add eslint-mdx@https://pkg.csb.dev/mdx-js/eslint-mdx/commit/d4a9ad99/eslint-mdx/_pkg.tgz eslint-plugin-mdx@https://pkg.csb.dev/mdx-js/eslint-mdx/commit/d4a9ad99/eslint-plugin-mdx/_pkg.tgz
# npm
npm i https://pkg.csb.dev/mdx-js/eslint-mdx/commit/d4a9ad99/eslint-mdx https://pkg.csb.dev/mdx-js/eslint-mdx/commit/d4a9ad99/eslint-plugin-mdx
```

**Comment by JounQin** [comment]
v2.2.1 has already been released.
